transcom/mymove
# B-21436: Allow AK to be entered as a state and Enable Alaska FF

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21436)
## [FF Repo PR](https://github.com/transcom/milmove-feature-flags/pull/210)

## Summary

In this PR I am allowing AK to be entered as a state. I am also creating the feature flag to enable or disable the ability to select AK as a state.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

*In all cases the message described in the testing steps should read "Moves to this state are not supported at this time".

1. Turn on enable Alaska feature flag
2. Access the MilMove application as a customer.
3. Login as a new user and when prompted for an address, try to select AK.
4. You should not see a message above the State field and the next button should not be disabled.
5. After selecting AK continue with customer info.
6. Next go to set up shipments. Here wherever address fields, including the optional address fields, select AK as a state. There should not be a message above the state field and the next button should not be disabled.
7. Next access office as a SC role.
8. Go to add a shipment.
9. Go through all shipment types selecting AK as the state. Should be able to move on without any messages denying AK as a state.
10. Turn off enable Alaska FF
11. Test again but you should not be able to select AK as a state.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots

AK is allowed to be entered as a state on Milmove:
![image](https://github.com/user-attachments/assets/dc57d623-4799-4f8e-bf58-b26f22a2c32f)

AK is allowed to be entered as an address in Office:
![image](https://github.com/user-attachments/assets/d83ba8b3-3d28-4cda-96bb-c8f73ca10971)

##508 Compliance Check
![image](https://github.com/user-attachments/assets/6fb44648-0ade-43b9-95d0-7d8543006ad4)
